<launch>
  <!-- only r2 is supported -->
  <arg name="sampling_space" value="r2" />

  <!-- load robot description -->
  <arg name="robot_model_path" value="$(find differentiable_rmap)/urdf/Simple2DoFManipulator.urdf.xacro" />
  <param name="robot_description" command="$(find xacro)/xacro $(arg robot_model_path)" />

  <node pkg="differentiable_rmap" type="NodeRmapSampling" name="rmap_sampling"
        output="screen">
    <rosparam subst_value="true" if="$(eval arg('sampling_space') == 'r2')">
      sampling_space: R2
      body_name: EEF
      joint_name_list: [Joint1, Joint2]
      bag_path: /tmp/rmap_sample_set_r2.bag
      sample_num: 10000
      sleep_rate: 2000
    </rosparam>
  </node>

  <node pkg="rviz" type="rviz" name="rviz"
        args="-d $(find differentiable_rmap)/rviz/RmapSamplingSimple2DoFManipulator.rviz"
        output="log" clear_params="true" />
</launch>
